experiment/autobumper/bumper/bumper.go

```diff
@@ -433,47 +426,44 @@ func updateReferences(imageBumperCli imageBumper, filterRegexp *regexp.Regexp, t
 }
 
 func upstreamImageVersionResolver(
-	upstreamVersionType string, parse func(upstreamAddress string) (string, error),
-) (func(imageHost, imageName, currentTag string) (string, error), error) {
-	prowUpstreamAddress, boskosUpstreamAddress, err := upstreamConfigFileAddresses(upstreamVersionType)
+	o *Options, upstreamVersionType string, parse func(upstreamAddress string) (string, error), imageBumperCli imageBumper) (func(imageHost, imageName, currentTag string) (string, error), error) {
+	upstreamVersions, err := upstreamConfigVersions(upstreamVersionType, o, parse)
 	if err != nil {
 		return nil, err
 	}
 
-	prowVersion, err := parse(prowUpstreamAddress)
-	if err != nil {
-		return nil, fmt.Errorf("error resolving the upstream Prow version from %q: %w", prowUpstreamAddress, err)
-	}
-	boskosVersion, err := parse(boskosUpstreamAddress)
-	if err != nil {
-		return nil, fmt.Errorf("error resolving the upstream Boskos version from %q: %w", boskosUpstreamAddress, err)
-	}
 	return func(imageHost, imageName, currentTag string) (string, error) {
-		imageFullPath := imageHost + "/" + imageName
-		if strings.HasPrefix(imageFullPath, prowPrefix) {
-			return prowVersion, nil
-		} else if strings.HasPrefix(imageFullPath, boskosPrefix) {
-			return boskosVersion, nil
-		} else {
-			return currentTag, nil
+		imageFullPath := imageHost + "/" + imageName + "/" + currentTag
```

Comment:
Since this is not actually a filepath, and the separator will always be "/", I don't want to use filepath.Join just in case it runs on an OS with a different separator. Was there another reason why I might want to use filepath?